services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -767,43 +809,40 @@ def compute_config_parquet_and_info_response(
         )
     except _EmptyDatasetError as err:
         raise EmptyDatasetError(f"{dataset=} is empty.", cause=err) from err
-    writer_batch_size = get_writer_batch_size(builder.info)
-    if writer_batch_size is not None and (
-        builder._writer_batch_size is None or builder._writer_batch_size > writer_batch_size
-    ):
-        builder._writer_batch_size = writer_batch_size
-    raise_if_too_big_from_external_data_files(
-        builder=builder,
-        max_dataset_size=max_dataset_size,
-        max_external_data_files=max_external_data_files,
-        hf_token=hf_token,
-    )
-    builder.download_and_prepare(file_format="parquet")  # the parquet files are stored in the cache dir
-    dataset_info = asdict(builder.info)
-    local_parquet_files.extend(
-        ParquetFile(local_file=local_file, local_dir=builder.cache_dir, config=config)
-        for local_file in glob.glob(f"{builder.cache_dir}**/*.parquet")
-    )
 
-    # create the target revision if it does not exist yet (clone from initial commit to avoid cloning all repo's files)
+    if is_parquet_builder_with_hub_files(builder, hf_endpoint=hf_endpoint):
+        parquet_operations = copy_parquet_files(builder)
+    else:
+        dataset_info = get_dataset_info_or_raise(
+            dataset=dataset, hf_endpoint=hf_endpoint, hf_token=hf_token, revision=source_revision
+        )
+        raise_if_not_supported(
+            dataset_info=dataset_info,
+            builder=builder,
+            hf_endpoint=hf_endpoint,
+            hf_token=hf_token,
+            supported_datasets=supported_datasets,
+            max_dataset_size=max_dataset_size,
+            max_external_data_files=max_external_data_files,
+        )
+        parquet_operations = convert_to_parquet(builder)
+
+    # create the target revision if we managed to get the parquet files and it does not exist yet
+    # (clone from initial commit to avoid cloning all repo's files)
     try:
-        refs = hf_api.list_repo_refs(repo_id=dataset, repo_type=DATASET_TYPE)
         if all(ref.ref != target_revision for ref in refs.converts):
             initial_commit = hf_api.list_repo_commits(repo_id=dataset, repo_type=DATASET_TYPE)[-1].commit_id
             committer_hf_api.create_branch(
-                repo_id=dataset, branch=target_revision, repo_type=DATASET_TYPE, revision=initial_commit
+                repo_id=dataset, branch=target_revision, repo_type=DATASET_TYPE, revision=initial_commit, exist_ok=True
```

Comment:
In case another job created the branch in the meantime. It's more likely to happen in this PR since I check the list of refs at the beginning (also to know if the repo exists) and only create the branch if the parquet conversion is successful.

Reply:
OK, good idea. Indeed, no need to raise an issue in that case